pkg/cmd/server/api/v1/types.go

```diff
@@ -138,6 +145,11 @@ type MasterConfig struct {
 
 	// NetworkConfig to be passed to the compiled in network plugin
 	NetworkConfig NetworkConfig `json:"networkConfig"`
+
+	// Enable or disable OpenShift components.
+	// TODO: find out how to write this field only if any of its field contains
```

Comment:
Thanks! `omitempty` is exactly what I was looking for. Each day learning new stuff about *go*.